pkg/detectors/sentrytoken/sentrytoken.go

```diff
@@ -76,54 +87,59 @@ func (s Scanner) FromData(ctx context.Context, verify bool, data []byte) (result
 	return results, nil
 }
 
-type Response []Project
-
-type Project struct {
-	Organization Organization `json:"organization"`
-}
-
-type Organization struct {
-	ID   string `json:"id"`
-	Name string `json:"name"`
-}
-
-func verifyToken(ctx context.Context, client *http.Client, token string) (bool, error) {
-	req, err := http.NewRequestWithContext(ctx, http.MethodGet, "https://sentry.io/api/0/projects/", nil)
+func verifyToken(ctx context.Context, client *http.Client, token string) (map[string]string, bool, error) {
+	// api docs: https://docs.sentry.io/api/organizations/
+	// this api will return 200 for user auth tokens with scope of org:<>
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, "https://sentry.io/api/0/organizations/", nil)
 	if err != nil {
-		return false, err
+		return nil, false, err
 	}
 	req.Header.Add("Authorization", fmt.Sprintf("Bearer %s", token))
 
-	res, err := client.Do(req)
+	resp, err := client.Do(req)
 	if err != nil {
-		return false, err
+		return nil, false, err
 	}
-	defer res.Body.Close()
+	defer func() {
+		_, _ = io.Copy(io.Discard, resp.Body)
+		_ = resp.Body.Close()
+	}()
 
-	var isVerified bool
-	switch res.StatusCode {
-	case http.StatusOK, http.StatusForbidden:
-		isVerified = true
-	case http.StatusUnauthorized:
-		return false, errUnauthorized
-	default:
-		return false, fmt.Errorf("unexpected HTTP response status %d", res.StatusCode)
+	bytes, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, false, err
 	}
 
-	bytes, readErr := io.ReadAll(res.Body)
-	if readErr != nil {
-		return false, readErr
-	}
+	switch resp.StatusCode {
+	case http.StatusOK:
+		var resp Response
+		if err = json.Unmarshal(bytes, &resp); err != nil {
+			return nil, false, err
+		}
 
-	var resp Response
-	if err = json.Unmarshal(bytes, &resp); err != nil {
-		return false, err
-	}
-	if len(resp) == 0 {
-		return false, fmt.Errorf("unexpected response body: %s", string(bytes))
-	}
+		var extraData = make(map[string]string)
+		for _, org := range resp {
+			extraData[fmt.Sprintf("orginzation_%s", org.ID)] = org.Name
+		}
+
+		return extraData, true, nil
+	case http.StatusForbidden:
+		var responseBody interface{}
+		if err := json.Unmarshal(bytes, &responseBody); err != nil {
+			return nil, false, err
+		}
 
-	return isVerified, err
+		// if response contain the forbiddenError message it means the token is active but does not have the right scope for this API call
+		if strings.Contains(fmt.Sprintf("%v", responseBody), forbiddenError) {
```

Comment:
In case the token is Active and does not have the `org:<>` scope the API returns 403 with a specific error message. In case token is removed the API return `401`